src/System.Management.Automation/engine/remoting/client/remoterunspace.cs

```diff
@@ -2005,27 +2005,210 @@ public override void StopProcessCommand()
             }
         }
 
-        /// <summary>
-        /// Adds the provided set of breakpoints to the debugger.
-        /// </summary>
-        /// <param name="breakpoints">Breakpoints.</param>
-        public override void SetBreakpoints(IEnumerable<Breakpoint> breakpoints)
-        {
-            _runspace.Debugger?.SetBreakpoints(breakpoints);
-        }
-
         /// <summary>
         /// Get a breakpoint by id, primarily for Enable/Disable/Remove-PSBreakpoint cmdlets.
         /// </summary>
         /// <param name="id">Id of the breakpoint you want.</param>
         public override Breakpoint GetBreakpoint(int id) =>
-            _runspace.Debugger?.GetBreakpoint(id);
+            InvokeVirtualBreakpointFunction<Breakpoint>(
+                DebuggerUtils.GetPSBreakpointFunctionName,
+                new Dictionary<string, object>
+                {
+                    { "Id", id },
+                });
 
         /// <summary>
         /// Returns breakpoints primarily for the Get-PSBreakpoint cmdlet.
         /// </summary>
-        public override List<Breakpoint> GetBreakpoints() =>
-            _runspace.Debugger?.GetBreakpoints();
+        public override List<Breakpoint> GetBreakpoints()
+        {
+            CheckForValidateState();
+
+            var breakpoints = new List<Breakpoint>();
+
+            using (PowerShell ps = GetNestedPowerShell())
+            {
+                ps.AddCommand(DebuggerUtils.GetPSBreakpointFunctionName);
+
+                Collection<PSObject> output = ps.Invoke<PSObject>();
+                foreach (var item in output)
+                {
+                    if (item?.BaseObject is Breakpoint)
+                    {
+                        breakpoints.Add((Breakpoint)item.BaseObject);
+                    }
+                }
+            }
+
+            return breakpoints;
+        }
+
+        /// <summary>
+        /// Adds the provided set of breakpoints to the remote debugger.
+        /// </summary>
+        /// <param name="breakpoints">Breakpoints.</param>
+        public override void SetBreakpoints(IEnumerable<Breakpoint> breakpoints)
```

Comment:
We should remove this method from the debugger.  It was only added to support Workflow debugging.  It was only partially implemented for local debugger, and since we have specific SetBreakpoint methods now I feel it is no longer necessary.  It is technically a breaking change, but I doubt anyone uses it and will be minimal impact.

Reply:
Removed and committed.

Reply:
@PaulHigin: What about PSThreadJob? Removing this will break that module.

Personally I don't mind that change, and these changes are part of why I want to update the PSThreadJob module with a few PRs: the one to make the Runspace property public (that you rejected, but that information is necessary for PowerShell users), and one to provide multiple versions of the binary module so that it works properly for PS 7+ and downlevel, with all of the functionality that is available in either case.

This is also why I want to create a new intermediate class that handles most of the wrapped debugger work: so that PSThreadJob's debugger, the PSTask debugger you just added, etc., can have a minimal implementation with the majority of the plumbing hooked up automatically, regardless of the version you are working with. That will allow for future changes to light-up additional functionality in the various debuggers that are mostly pass-thru at no cost.

Reply:
PSThreadJob hasn't been published with the IJobDebugger change, so it can still be easily changed.  It only implements SetBreakpoints() because Tyler added that to the Debugger base class ... it is not needed and can be safely removed.  Same goes for PSTaskJob.

The IJobDebugger was added to deal with WSMan based remote jobs to handle common data throttling problems, and is not needed for threadjob and taskjob jobs.  So I agree that the code needs to be re-thought, but that can be done separately from this work.